src/Runner.Worker/FileCommandManager.cs

```diff
@@ -383,6 +395,34 @@ public EnvFileKeyValuePairs(IExecutionContext context, string filePath)
             return GetEnumerator();
         }
 
+        // accepts a string, two indexes, and returns true if only whitespace chars exist between those two indexes (non-inclusive)
+        private static bool CheckIfOnlyWhitespaceBetweenPositions(string str, int pos1, int pos2)
+        {
+            // Check if the provided positions are valid
+            if (pos1 < 0 || pos2 < 0 || pos1 >= str.Length || pos2 >= str.Length)
+            {
+                throw new ArgumentOutOfRangeException("Whitespace check: Positions should be within the string length.");
+            }
+
+            // Ensure pos1 is always the smaller position
+            if (pos2 < pos1)
+            {
+                int temp = pos2;
+                pos2 = pos1;
+                pos1 = temp;
+            }
```

Comment:
I am not sure we should flip these. If equalsIndex is after heredocIndex, then we have a string like:

```
FOO << =EOF
abc
=EOF
```

I think we can just check `if pos2 < pos1` and throw an argument exception.